backend/frontend/tests/test_progress_details_pages.py

```diff
@@ -307,3 +310,63 @@ def test_progress_details_displays_correct_details_for_topic(
             self.topic.content_url,
             body,
         )
+
+
+class TestRepoProjectDetailsPage(FrontendTestMixin):
+    def setUp(self):
+        super().setUp()
+        self.user = UserFactory(email="learner_1@umuzi.org")
+        self.user.set_password(self.user.email)
+        self.user.save()
+        self.do_login(self.user)
+
+    def _get_project_progress_details_url(self):
+        return self.reverse_url(
+            PROGRESS_DETAILS_VIEW,
+            kwargs={
+                "content_type": "project",
+                "id": self.card.id,
+            },)
+    
+    def make_ip_project_card(self):
+        self.card = AgileCardFactory()
+        self.card.recruit_project.repository = RepositoryFactory(user=self.user)
+        self.card.recruit_project.save()
+        self.card.save()
+
+        self.card.assignees.set([self.user])
+        self.card.recruit_project.recruit_users.set([self.user])
+
+    def test_progress_details_page_displays_repo_for_repo_project(self):
+        self.make_ip_project_card()
+
+        self.repo_project_progress_details_url = self._get_project_progress_details_url()
+
```

Comment:
Much better now, but i still think attaching this to "self" is not necessary since no other part of our code outside this function is going to use it.

Reply:
Oh I see what you mean